File: front-ParkWise/src/app/models/claim.ts
import { ResponseClaim } from "./responseClaim";
import { Status } from "./status";
import { User } from "./user";


export class Claim {
    id!: number;
    user!: User; 
    subject!: string;
    description!: string;
    createdAt!: Date;
    status!: Status;
    claimResponse?:ResponseClaim;
  }

File: backoffice/src/main/java/com/example/achref/Entities/claims/Claim.java
package com.example.achref.Entities.claims;

import com.example.achref.Entities.user.User;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Getter
@Setter
public class Claim implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    private String subject;

    @Column(columnDefinition = "TEXT")
    private String description;

    private LocalDateTime createdAt;
    @Enumerated(EnumType.STRING)
    private Status status;

    @OneToOne(mappedBy="claim",cascade = CascadeType.ALL,orphanRemoval = true)
    @JsonIgnore
    private ClaimResponse response;
}
